packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -148,6 +148,41 @@ library LibVerifying {
         reward = (reward * (10000 - config.rewardBurnBips)) / 10000;
     }
 
+    function getProverRewardBips(
+        uint256 proverRewardRandomizedPercentage,
+        uint256 numProvers
+    ) public view returns (uint256[] memory weights) {
+        uint256 randomized = proverRewardRandomizedPercentage;
+        if (randomized > 100) {
+            randomized = 100;
+        }
+
+        uint256 sum;
+        uint256 i;
+
+        if (randomized > 0) {
+            unchecked {
+                // calculate the randomized weight
+                uint256 seed = block.prevrandao;
+                for (i = 0; i < numProvers; ++i) {
+                    weights[i] = uint16(seed * (1 + i));
```

Comment:
```
weights[i] = uint16(seed * (1 + i));
```
Not sure if this is a good way to create a random uint16 for each prover.

Reply:
i think it looks OK.